gammapy/time/lightcurve.py

```diff
@@ -320,6 +321,136 @@ def create_fixed_time_bin(time_step, spectrum_extraction):
             ])
         return intervals
 
+    def create_fixed_significance_bin_lc_table(self, significance, significance_method, energy_range,
+                                               spectrum_extraction, separators=[]):
+        """
+
+        Create time intervals such that each bin of a light curve reach a given significance and store information
+
+        Parameters
+        ----------
+        significance : float
+            Target significance for each light curve point
+        significance_method : {‘lima’, ‘simple’}
+            Significance method (see `~gammapy.stats.significance_on_off`)
+        energy_range : `~astropy.units.Quantity`
+            True energy range to evaluate integrated flux (true energy)
+        spectrum_extraction : `~gammapy.spectrum.SpectrumExtraction`
+            Contains statistics, IRF and event lists
+        separators : `list` of float
+            Contains a list of time (in MJD) to break the current point creation
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table of time intervals  and information about their content : on/off events, alpha, significance
+
+        Examples
+        -------
+        extract intervals for light curve :
+            intervals = list(zip(table['t_start'], table['t_stop']))
+
+        """
+
+        def in_list(item, L):
+            o, j = np.where(L == item)
+            for index in o:
+                yield index
+
+        time_holder = []
+        obs_properties = []
+        n_obs = 0
+
+        for time in separators:
+            time_holder.append([Time(time,format('mjd')), 'break'])
+
+        for obs in spectrum_extraction.obs_list:
+            # start and end will need to be redefined once the data storage format is fixed
+            time_holder.append([obs.events.obs_start, 'start'])
+            time_holder.append([obs.events.obs_stop, 'end'])
+            obs_properties.append(dict(deadtime=obs.observation_dead_time_fraction,
+                                       A_off=spectrum_extraction.bkg_estimate[n_obs].a_off))
+            n_obs += 1
+        obs_properties = Table(rows=obs_properties)
+
+        # prepare the on and off photon list as in the flux point computation -> should be updated accordingly
+        for t_index, obs in enumerate(self.obs_list):
+            spec = self.obs_spec[t_index]
+            # get ON and OFF evt list
+            off_evt = self.off_evt_list[t_index]
+            on_evt = self.on_evt_list[t_index]
+            # prepare energy range filter
+            e_reco = spec.e_reco
+            emin = e_reco[e_reco.searchsorted(max(spec.lo_threshold, energy_range[0]))]
+            emax = e_reco[e_reco.searchsorted(min(spec.hi_threshold, energy_range[1])) - 1]
+            # filter ON events
+            on = on_evt.select_energy([emin, emax])
+            on = on.select_energy(energy_range)
```

Comment:
Is it necessary to call `select_energy` twice ?

Reply:
I don't think so seeing the definition of emin and emax. But since I'm not sure I keeped what is in place